prow/flagutil/config.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+// ConfigOptions holds options for interacting with Configs.
+type ConfigOptions struct {
+	EnableConfig     bool
+	ConfigPath       string
+	JobConfigPath    string
+	PluginConfigPath string
+}
+
+// AddFlags injects config options into the given FlagSet.
+func (o *ConfigOptions) AddFlags(fs *flag.FlagSet) {
+	fs.BoolVar(&o.EnableConfig, "enable-config", true, "Enable config options, set false to disable config options.")
```

Comment:
What is this?

Reply:
This was added to allow disabling of `os.Stat` checking during config validation. I welcome a better idea, this was what I came up with :smile:

Reply:
Whether or not we do validation should not be up to the user so I don't think this should be a flag. We may do different types of validation for different binaries, but the validation behavior should be fixed per binary, right?